packages/icons/tasks/scaffold-metadata.js

```diff
@@ -68,13 +79,50 @@ async function scaffold() {
     {}
   );
 
+  /**
+   * build object that maps icon names to category & subcategory
+   *   - loop through all of `categories.yml`
+   *   - keys are icon names
+   *   - each iconName object has props for category & subcategory
+   *
+   * when needing to find category information,
+   * `categoryInformation[iconName].category` and
+   * `categoryInformation[iconName].subcategory` will return needed info
+   */
+  let categoryInformation = {};
+  categoriesJson['categories'].forEach(category => {
+    category.subcategories.forEach(subcategory => {
+      subcategory.members.forEach(iconName => {
+        categoryInformation[iconName] = {
+          category: category.name,
+          subcategory: subcategory.name,
+        };
+      });
+    });
+  });
+
   const icons = Object.keys(iconsGroupedByName).map(key => {
+    let iconIsCategorized =
+      categoryInformation[key] &&
+      categoryInformation[key].hasOwnProperty('subcategory');
+
+    // this warning will still be called if all "variants" are defined but there is no "main"
+    // may be a worthwhile warning if we check the variants tho?
+    // if (!iconIsCategorized) {
+    //   console.log(`warning: "${key}" is not defined in categories.yml.`);
+    // }
+
     const group = iconsGroupedByName[key];
     const icon = {
       name: key,
       friendly_name: sentenceCase(key),
       usage: 'This is a description for usage',
-      categories: ['example cateogry'],
+      category: iconIsCategorized
+        ? categoryInformation[key].category
+        : 'uncategorized',
```

Comment:
Instead of using `uncategorized`, should we do empty fields? In other words, empty category/subcategory fields in `metadata.yml`

Reply:
@joshblack maybe "other" ? i was trying to think of something that would work as a catch-all for the menu in sketch, so all uncategorized icons could under an "other" or "uncategorized" menu item

Reply:
however that could be implemented without actually storing that in the metadata so yeah i get your point…